internal/reminder/check_test.go

```diff
@@ -0,0 +1,160 @@
+package reminder_test
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"hellper/internal/bot"
+	"hellper/internal/config"
+	"hellper/internal/log"
+	"hellper/internal/model"
+	"hellper/internal/reminder"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/slack-go/slack"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type checkReminderFixture struct {
+	testName       string
+	expected       bool
+	ctx            context.Context
+	mockLogger     log.Logger
+	mockClient     bot.Client
+	mockRepository model.Repository
+	mockIncident   model.Incident
+
+	channelID     string
+	listPinsError error
+	lastPin       []slack.Item
+}
+
+func (f *checkReminderFixture) setup(t *testing.T) {
+	var (
+		loggerMock     = log.NewLoggerMock()
+		clientMock     = bot.NewClientMock()
+		repositoryMock = model.NewRepositoryMock()
+	)
+
+	f.ctx = context.Background()
+
+	loggerMock.On("Info", f.ctx, mock.AnythingOfType("string"), mock.AnythingOfType("[]log.Value")).Return()
+	clientMock.On("ListPins", mock.AnythingOfType("string")).Return(f.lastPin, nil, nil)
+
+	f.mockLogger = loggerMock
+	f.mockClient = clientMock
+	f.mockRepository = repositoryMock
+
+}
+func TestCanSendNotify(t *testing.T) {
+
+	table := []checkReminderFixture{
```

Comment:
A good test here but made me think, could we structure our fixtures to be more clear the scenarios? I though in something like RSpec describes, where you can separate similar scenarios from different ones. I don't know yet how to do this, but I think is something to think of.

Reply:
Nice! I couldn't understand what it would be like. Can we refactor it in a v2?

Reply:
Totally, was not a request to change here, but a provocation to ourselves about how to write better tests scenarios